embrace zstm

mschuwalow opened the pull request:
No logic changes, just cleanup

mijicd commented on membership/src/main/scala/zio/membership/hyparview/package.scala:
```diff
@@ -132,75 +132,68 @@ package object hyparview {
-                                            .mapError(e => zio.membership.SerializationError(e.msg))
-                                  _ <- con.send(reply)
-                                } yield Some(msg.sender)
+                                  full <- Views.isActiveViewFull
```
We have `ifM` as well :).

mijicd commented on membership/src/main/scala/zio/membership/hyparview/package.scala:
```diff
@@ -228,17 +221,14 @@ package object hyparview {
-              }
+            ZSTM.atomically {
+              for {
+                _              <- Views.isActiveViewFull.flatMap(full => STM.check(!full))
```
You can use `retryWhile` or `retryUntil` here.

mijicd commented on membership/src/main/scala/zio/membership/hyparview/periodic.scala:
```diff
@@ -9,48 +9,43 @@ import zio.logging.Logging.Logging
+          }
+          .commit
+          .flatten
+      go.eventually
```
You can remove val here :).

mijicd commented on membership/src/main/scala/zio/membership/hyparview/periodic.scala:
```diff
@@ -9,48 +9,43 @@ import zio.logging.Logging.Logging
+          .map(_.toList)
+          .flatMap { nodes =>
+            TRandom.selectOne(nodes).flatMap {
+              case None => STM.succeed(Views.viewState.commit)
```
Why not `Views.viewState` only?

mschuwalow replied:
It expects an STM[_, UIO[_]], so I rolled with this one

mschuwalow commented:
@mijicd thanks for your review. I think I addressed most of the points, can you take another look?

I'm also running into flakiness with the tests as they are currently using a fixed port for the TCPTransportSpec. Sadly I can't think of a good way to fix that without polluting our interfaces. Maybe you can take a look?
